src/server.h

```diff
@@ -3336,15 +3340,19 @@ robj *setTypeDup(robj *o);
 #define HASH_SET_COPY 0
 
 
+void dbUpdateKeyWithVolaItemsTracking(serverDb *db, robj *o);
 void hashTypeFreeVolatileSet(robj *o);
-void hashTypeTrackEntry(robj *o, void *entry);
-void hashTypeUntrackEntry(robj *o, void *entry);
-void hashTypeTrackUpdateEntry(robj *o, void *old_entry, void *new_entry, long long old_expiry, long long new_expiry);
+void hashTypeTrackEntry(serverDb *db, robj *o, void *entry);
+void hashTypeUntrackEntry(serverDb *db, robj *o, void *entry);
+void hashTypeTrackUpdateEntry(serverDb *db, robj *o, void *old_entry, void *new_entry, long long old_expiry, long long new_expiry);
```

Comment:
As I noted elsewhere, I think that all 3 of these APIs should remain unchanged.  The hashtype (in t_hash) should be independent of the tracking in the DB (db.c).  We should be able to have a disconnected hash type robj.

IMO these 3 APIs were correctly working on an independent robj (independent of being in the main dictionary).